packages/components/src/internal/components/domainproperties/DomainRow.tsx

```diff
@@ -532,9 +572,27 @@ export class DomainRow extends React.PureComponent<DomainRowProps, DomainRowStat
                                 />
                             </div>
                         </Collapse>
+                        {dataTypeChangeToConfirm && (
+                            <ConfirmDataTypeChangeModal
+                                originalRangeURI={field.original.rangeURI}
+                                newDataType={PropDescType.fromName(dataTypeChangeToConfirm)}
+                                onConfirm={this.onConfirmTypeChange}
+                                onCancel={this.onHideConfirmTypeChange}
+                            />
+                        )}
                     </div>
                 )}
             </Draggable>
         );
     }
 }
+
+const shouldShowConfirmDataTypeChange = (originalRangeURI: string, newRangeURI: string): boolean => {
+    if (newRangeURI && originalRangeURI !== newRangeURI) {
+        const wasString = STRING_CONVERT_URIS.indexOf(originalRangeURI) > -1;
+        const toString = STRING_CONVERT_URIS.indexOf(newRangeURI) > -1;
+        const toNumber = NUMBER_CONVERT_URIS.indexOf(newRangeURI) > -1;
+        return toNumber || (toString && !wasString);
```

Comment:
I don't see how this would flag boolean->string for a warning

Reply:
For a field saved as a boolean, the `originalRangeURI` would be `http://www.w3.org/2001/XMLSchema#boolean` and the `newRangeURI` would be `http://www.w3.org/2001/XMLSchema#string`. There for the original would say that it is not `wasString` and the new would say that it is going `toString`.

Reply:
Ah, right. Thanks